### DISCUSSION TITLE
Provide SQL-Server as resource to custom image

### DISCUSSION THREAD

**Discussion by gkfischer** [microsoft/aspire#1912] [Q&A]
After reading a couple of Issues and discussions, I think it's currently impossible to provide a resource like a SQL Server, created with `AddSqlServerContainer()`, to a custom image, loaded with `AddContainer`. 

The scenario for this is a micro-service landscape, where a system being developed e.g. Accounting, is dependent on another microservice, e.g. User Authorization, developed by a different team. The Accounting service also needs an endpoint to talk to the User Management. It sounds exactly like a problem that Aspire is supposed to solve, right? Tying them together in the source code is no reasonable option. Another example is a RabbitMQ container required by both services.

As far as I can see, there's an unresolved issue around networking in docker prohibiting this, so I've got two questions:
  1. is this the current state of affairs, or did I miss something altogether and this is already possible?
  2. will there be a solution for this scenario in one of the next releases?

**Comment by davidfowl** [comment]
> After reading a couple of Issues and discussions, I think it's currently impossible to provide a resource like a SQL Server, created with AddSqlServerContainer(), to a custom image, loaded with AddContainer.

I'm confused about this. What exactly does this mean?

What code do you want to write? What did you attempt to do already for this scenario?

**Comment by gkfischer** [comment]
Sorry for the delayed answer. I'll try to explain the scenario I would like to use.

In a microservice environment with multiple teams working on different services, I have something like this:

Team A is creating a user management using SQL-Server as storage and RabbitMQ as a message bus
Team B is creating an ordering module also using SQL-Server and RabbitMQ and is relying on the user management

What I would like to do when developing the ordering module is to use a stable docker image of the user management and not having to check out and build the source code of it. 

Using Aspire I can spin up the SQL-Server, the RabbitMQ, and the user management container. 

But how would I supply the SQL-Server and RabbitMQ resources to the user management container?

**Reply by davidfowl** [reply]
> What I would like to do when developing the ordering module is to use a stable docker image of the user management and not having to check out and build the source code of it.

So you are using containers from another team and you want to spin up their dependencies and their service right?

**Reply by gkfischer** [reply]
Yes, exactly

**Comment by robertcoltheart** [comment]
I'm also interested in a solution that let's us tailor Aspire to a microservice environment. Ie. Service A requires MSSQL, and Service B depends on Service A (developed by possibly different teams) and it's dependencies transitively.

At the moment the solution we have landed on is to publish a Nuget package in Service A that lets you call `.AddServiceA()` which will wire up MSSQL, while also checking the existing resources for an existing MSSQL resource (since we only want one MSSQL running in an AppHost). Service B can import this Nuget and wire up Service A and its dependencies behind the scenes.

However there are a lot of rough edges with this. For example, what if Service B requires MSSQL but with a slightly different configuration or set of users. Or what if we actually _do_ want to run 2 instances of MSSQL. We don't really have answers for these edge cases yet, other than a lot of very ugly parameter-passing or `Action<ServiceAOptions>` delegate parameters.
